manifests/config.pp

```diff
@@ -77,4 +78,13 @@
   file { '/var/lib/candlepin/.puppet-candlepin-rpm-version':
     ensure => absent,
   }
+
+  if $candlepin::facts_match_regex {
```

Comment:
This shouldn't be needed as you already added to the context here - https://github.com/theforeman/puppet-candlepin/pull/262/files#diff-b55cd2d4fee4ad88a9e21316ff2015c5579e8ec9bef7bc0fe5253262905fa702R33 and you included the line in the template.

You'll need to add it to the top of the template as a parameter in this section: https://github.com/theforeman/puppet-candlepin/pull/262/files#diff-fe2528baf26c583efc94c2a9d20332fd9d4cf0b0ca46f8e7201b12bbc3a139d1R17

And you can add a spec unit test to check that passing in the parameter does the right thing, similar to: https://github.com/theforeman/puppet-candlepin/blob/master/spec/classes/candlepin_spec.rb#L286-L297